# PyCity Schools Analysis

* As a whole, schools with higher budgets, did not yield better test results. By contrast, schools with higher spending per student actually (\$645-675) underperformed compared to schools with smaller budgets (<\$585 per student).

* As a whole, smaller and medium sized schools dramatically out-performed large sized schools on passing math performances (89-91% passing vs 67%).

* As a whole, charter schools out-performed the public district schools across all metrics. However, more analysis will be required to glean if the effect is due to school practices or the fact that charter schools tend to serve smaller student populations per school. 
---

### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [2]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
school_data_to_load = "Resources/schools_complete.csv"
student_data_to_load = "Resources/students_complete.csv"

# Read School and Student Data File and store into Pandas Data Frames
school_data = pd.read_csv(school_data_to_load)
student_data = pd.read_csv(student_data_to_load)

# Combine the data into a single dataset
school_data_complete = pd.merge(student_data, school_data, how="left", on=["school_name"])
school_data_complete.head()

Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  School ID      type  size   budget  
0             66          79          0  District  2917  1910635  
1             94          61          0  District  2917  1910635  
2             90          60          0  District  2917  1910635  
3             67          58          0  District  2917  1910635  
4             97          84          0  District  2917  1910635

## District Summary

* Calculate the total number of schools

* Calculate the total number of students

* Calculate the total budget

* Calculate the average math score 

* Calculate the average reading score

* Calculate the overall passing rate (overall average score), i.e. (avg. math score + avg. reading score)/2

* Calculate the percentage of students with a passing math score (70 or greater)

* Calculate the percentage of students with a passing reading score (70 or greater)

* Create a dataframe to hold the above results

* Optional: give the displayed data cleaner formatting

In [3]:
#Calculate metrics
#Number of schools
num_schools=school_data_complete['school_name'].nunique()

#Number of students
num_students=school_data_complete['student_name'].count()

#Total Budget
budget_schools=school_data_complete['budget'].unique().sum()

#Average math score
avg_math_score=school_data_complete['math_score'].mean()

#Average reading score
avg_reading_score=school_data_complete['reading_score'].mean()

#Passing Rate
pass_rate=(avg_math_score+avg_reading_score)/2

#Math score percentage
num_pass_math = len(school_data_complete.loc[school_data_complete["math_score"] >= 70, :])
percent_pass_math=num_pass_math/num_students

#Reading Score Percentage
num_pass_reading=len(school_data_complete.loc[school_data_complete["reading_score"] >= 70, :])
percent_pass_reading=num_pass_reading/num_students

#Create District Summary Data Frame
dist_summ_df = pd.DataFrame({"Total Schools": [num_schools],
                                "Total Students": [num_students],
                                "Total Budget": [budget_schools],
                                "Average Math Score": [avg_math_score],
                                "Average Reading Score": [avg_reading_score],
                                "% Passing Math": [percent_pass_math],
                                "% Passing Reading": [percent_pass_reading],
                                "% Overall Passing Rate": [pass_rate]})

#Format Data Frame
dist_summ_df_clean=dist_summ_df
dist_summ_df_clean['Total Students']=dist_summ_df["Total Students"].map("{:,}".format)
dist_summ_df_clean['Total Budget']=dist_summ_df["Total Budget"].map("${:,.2f}".format)
dist_summ_df_clean['Average Math Score']=dist_summ_df["Average Math Score"].map("{:.2f}%".format)
dist_summ_df_clean['Average Reading Score']=dist_summ_df["Average Reading Score"].map("{:.2f}%".format)
dist_summ_df_clean['% Passing Math']=dist_summ_df["% Passing Math"].map("{:.2%}".format)
dist_summ_df_clean['% Passing Reading']=dist_summ_df["% Passing Reading"].map("{:.2%}".format)
dist_summ_df_clean['% Overall Passing Rate']=dist_summ_df["% Overall Passing Rate"].map("{:.2f}%".format)

dist_summ_df_clean
     
      
     
    

Total Schools Total Students    Total Budget Average Math Score  \
0             15         39,170  $24,649,428.00             78.99%   

  Average Reading Score % Passing Math % Passing Reading  \
0                81.88%         74.98%            85.81%   

  % Overall Passing Rate  
0                 80.43%

## School Summary

* Create an overview table that summarizes key metrics about each school, including:
  * School Name
  * School Type
  * Total Students
  * Total School Budget
  * Per Student Budget
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * Overall Passing Rate (Average of the above two)
  
* Create a dataframe to hold the above results

## Top Performing Schools (By Passing Rate)

* Sort and display the top five schools in overall passing rate

In [4]:
#Group by schools
by_school_group=school_data_complete.groupby('school_name')

#Calculate Metrics
#School Type
school_type=by_school_group['type'].first()

#Number of students by school
num_students_school=by_school_group['student_name'].count()

#Budget by school
budget_schools_indv=by_school_group['budget'].first()

#Budget per student
budget_student=budget_schools_indv/num_students_school

#Average math score by schoool
avg_math_school=by_school_group['math_score'].mean()

#Average reading score by school
avg_reading_school=by_school_group['reading_score'].mean()

# Math score percentage by school
num_pass_math_school = school_data_complete.loc[school_data_complete['math_score']>=70].groupby(['school_name']).size()
percent_pass_math_school=num_pass_math_school/num_students_school

#Reading Score Percentage by school
num_pass_reading_school=school_data_complete.loc[school_data_complete['reading_score']>=70].groupby(['school_name']).size()
percent_pass_reading_school=num_pass_reading_school/num_students_school

#Passing Rate by school
pass_rate_school=(percent_pass_math_school+percent_pass_reading_school)/2

#Create School Data Frame
school_summ_df = pd.DataFrame({"School Type": school_type,
                                "Total Students": num_students_school,
                                "Total School Budget": budget_schools_indv,
                                "Per Student Budget": budget_student,
                                "Average Math Score": avg_math_school,
                                "Average Reading Score": avg_reading_school,
                                "% Passing Math": percent_pass_math_school,
                                "% Passing Reading": percent_pass_reading_school,
                                "% Overall Passing Rate": pass_rate_school})

#Format Data Frame Top 5
school_summ_df_clean=school_summ_df
school_summ_df_clean.index.name = None
school_summ_df_clean = school_summ_df.sort_values(by = "% Overall Passing Rate", ascending=False).head(5)
school_summ_df_clean['Total Students']=school_summ_df['Total Students'].map('{:,}'.format)
school_summ_df_clean['Total School Budget']=school_summ_df['Total School Budget'].map('${:,.2f}'.format)
school_summ_df_clean['Per Student Budget']=school_summ_df['Per Student Budget'].map('${:,.2f}'.format)
school_summ_df_clean['Average Math Score']=school_summ_df['Average Math Score'].map('{:.2f}%'.format)
school_summ_df_clean['Average Reading Score']=school_summ_df['Average Reading Score'].map('{:.2f}%'.format)
school_summ_df_clean['% Passing Math']=school_summ_df['% Passing Math'].map('{:.2%}'.format)
school_summ_df_clean['% Passing Reading']=school_summ_df['% Passing Reading'].map('{:.2%}'.format)
school_summ_df_clean['% Overall Passing Rate']=school_summ_df['% Overall Passing Rate'].map('{:.2%}'.format)

school_summ_df_clean


School Type Total Students Total School Budget  \
Cabrera High School     Charter          1,858       $1,081,356.00   
Thomas High School      Charter          1,635       $1,043,130.00   
Pena High School        Charter            962         $585,858.00   
Griffin High School     Charter          1,468         $917,500.00   
Wilson High School      Charter          2,283       $1,319,574.00   

                    Per Student Budget Average Math Score  \
Cabrera High School            $582.00             83.06%   
Thomas High School             $638.00             83.42%   
Pena High School               $609.00             83.84%   
Griffin High School            $625.00             83.35%   
Wilson High School             $578.00             83.27%   

                    Average Reading Score % Passing Math % Passing Reading  \
Cabrera High School                83.98%         94.13%            97.04%   
Thomas High School                 83.85%         93.27%            97.31%   
Pena High School                   84.04%         94.59%            95.95%   
Griffin High School                83.82%         93.39%            97.14%   
Wilson High School                 83.99%         93.87%            96.54%   

                    % Overall Passing Rate  
Cabrera High School                 95.59%  
Thomas High School                  95.29%  
Pena High School                    95.27%  
Griffin High School                 95.27%  
Wilson High School                  95.20%

## Bottom Performing Schools (By Passing Rate)

* Sort and display the five worst-performing schools

In [5]:
#Group by schools
by_school_group=school_data_complete.groupby('school_name')

#Calculate Metrics
#School Type
school_type=by_school_group['type'].first()

#Number of students by school
num_students_school=by_school_group['student_name'].count()

#Budget by school
budget_schools_indv=by_school_group['budget'].first()

#Budget per student
budget_student=budget_schools_indv/num_students_school

#Average math score by schoool
avg_math_school=by_school_group['math_score'].mean()

#Average reading score by school
avg_reading_school=by_school_group['reading_score'].mean()

# Math score percentage by school
num_pass_math_school = school_data_complete.loc[school_data_complete['math_score']>=70].groupby(['school_name']).size()
percent_pass_math_school=num_pass_math_school/num_students_school

#Reading Score Percentage by school
num_pass_reading_school=school_data_complete.loc[school_data_complete['reading_score']>=70].groupby(['school_name']).size()
percent_pass_reading_school=num_pass_reading_school/num_students_school

#Passing Rate by school
pass_rate_school=(percent_pass_math_school+percent_pass_reading_school)/2

#Create School Data Frame
school_summ_df = pd.DataFrame({"School Type": school_type,
                                "Total Students": num_students_school,
                                "Total School Budget": budget_schools_indv,
                                "Per Student Budget": budget_student,
                                "Average Math Score": avg_math_school,
                                "Average Reading Score": avg_reading_school,
                                "% Passing Math": percent_pass_math_school,
                                "% Passing Reading": percent_pass_reading_school,
                                "% Overall Passing Rate": pass_rate_school})

#Format Data Frame Bottom 5
school_summ_df_clean=school_summ_df
school_summ_df_clean.index.name = None
school_summ_df_clean = school_summ_df.sort_values(by = "% Overall Passing Rate").head(5)
school_summ_df_clean['Total Students']=school_summ_df['Total Students'].map('{:,}'.format)
school_summ_df_clean['Total School Budget']=school_summ_df['Total School Budget'].map('${:,.2f}'.format)
school_summ_df_clean['Per Student Budget']=school_summ_df['Per Student Budget'].map('${:,.2f}'.format)
school_summ_df_clean['Average Math Score']=school_summ_df['Average Math Score'].map('{:.2f}%'.format)
school_summ_df_clean['Average Reading Score']=school_summ_df['Average Reading Score'].map('{:.2f}%'.format)
school_summ_df_clean['% Passing Math']=school_summ_df['% Passing Math'].map('{:.2%}'.format)
school_summ_df_clean['% Passing Reading']=school_summ_df['% Passing Reading'].map('{:.2%}'.format)
school_summ_df_clean['% Overall Passing Rate']=school_summ_df['% Overall Passing Rate'].map('{:.2%}'.format)

school_summ_df_clean


School Type Total Students Total School Budget  \
Rodriguez High School    District          3,999       $2,547,363.00   
Figueroa High School     District          2,949       $1,884,411.00   
Huang High School        District          2,917       $1,910,635.00   
Johnson High School      District          4,761       $3,094,650.00   
Ford High School         District          2,739       $1,763,916.00   

                      Per Student Budget Average Math Score  \
Rodriguez High School            $637.00             76.84%   
Figueroa High School             $639.00             76.71%   
Huang High School                $655.00             76.63%   
Johnson High School              $650.00             77.07%   
Ford High School                 $644.00             77.10%   

                      Average Reading Score % Passing Math % Passing Reading  \
Rodriguez High School                80.74%         66.37%            80.22%   
Figueroa High School                 81.16%         65.99%            80.74%   
Huang High School                    81.18%         65.68%            81.32%   
Johnson High School                  80.97%         66.06%            81.22%   
Ford High School                     80.75%         68.31%            79.30%   

                      % Overall Passing Rate  
Rodriguez High School                 73.29%  
Figueroa High School                  73.36%  
Huang High School                     73.50%  
Johnson High School                   73.64%  
Ford High School                      73.80%

## Math Scores by Grade

* Create a table that lists the average Reading Score for students of each grade level (9th, 10th, 11th, 12th) at each school.

  * Create a pandas series for each grade. Hint: use a conditional statement.
  
  * Group each series by school
  
  * Combine the series into a dataframe
  
  * Optional: give the displayed data cleaner formatting

In [6]:
#9th Grade math scores
grade9_math= school_data_complete[school_data_complete['grade']=='9th'].groupby('school_name')['math_score'].mean()

#10th Grade math scores
grade10_math= school_data_complete[school_data_complete['grade']=='10th'].groupby('school_name')['math_score'].mean()

#11th Grade math scores
grade11_math= school_data_complete[school_data_complete['grade']=='11th'].groupby('school_name')['math_score'].mean()

#12th Grade math scores
grade12_math= school_data_complete[school_data_complete['grade']=='12th'].groupby('school_name')['math_score'].mean()

#Create Math Scores Data Frame
math_grades_df = pd.DataFrame({"9th": grade9_math,
                                "10th": grade10_math,
                                "11th": grade11_math,
                                "12th": grade12_math,})

#Format Data Frame
math_grades_df_clean=math_grades_df
math_grades_df_clean.index.name= None
math_grades_df_clean['9th']=math_grades_df['9th'].map('{:.2f}%'.format)
math_grades_df_clean['10th']=math_grades_df['10th'].map('{:.2f}%'.format)
math_grades_df_clean['11th']=math_grades_df['11th'].map('{:.2f}%'.format)
math_grades_df_clean['12th']=math_grades_df['12th'].map('{:.2f}%'.format)
math_grades_df_clean


9th    10th    11th    12th
Bailey High School     77.08%  77.00%  77.52%  76.49%
Cabrera High School    83.09%  83.15%  82.77%  83.28%
Figueroa High School   76.40%  76.54%  76.88%  77.15%
Ford High School       77.36%  77.67%  76.92%  76.18%
Griffin High School    82.04%  84.23%  83.84%  83.36%
Hernandez High School  77.44%  77.34%  77.14%  77.19%
Holden High School     83.79%  83.43%  85.00%  82.86%
Huang High School      77.03%  75.91%  76.45%  77.23%
Johnson High School    77.19%  76.69%  77.49%  76.86%
Pena High School       83.63%  83.37%  84.33%  84.12%
Rodriguez High School  76.86%  76.61%  76.40%  77.69%
Shelton High School    83.42%  82.92%  83.38%  83.78%
Thomas High School     83.59%  83.09%  83.50%  83.50%
Wilson High School     83.09%  83.72%  83.20%  83.04%
Wright High School     83.26%  84.01%  83.84%  83.64%

## Reading Score by Grade 

* Perform the same operations as above for reading scores

In [7]:
#9th Grade reading scores
grade9_reading= school_data_complete[school_data_complete['grade']=='9th'].groupby('school_name')['reading_score'].mean()

#10th Grade reading scores
grade10_reading= school_data_complete[school_data_complete['grade']=='10th'].groupby('school_name')['reading_score'].mean()

#11th Grade reading scores
grade11_reading= school_data_complete[school_data_complete['grade']=='11th'].groupby('school_name')['reading_score'].mean()

#12th Grade reading scores
grade12_reading= school_data_complete[school_data_complete['grade']=='12th'].groupby('school_name')['reading_score'].mean()

#Create Math reading Data Frame
reading_grades_df = pd.DataFrame({"9th": grade9_reading,
                                "10th": grade10_reading,
                                "11th": grade11_reading,
                                "12th": grade12_reading,})


# #Format Data Frame
reading_grades_df_clean=reading_grades_df
reading_grades_df_clean.index.name= None
reading_grades_df_clean['9th']=reading_grades_df['9th'].map('{:.2f}%'.format)
reading_grades_df_clean['10th']=reading_grades_df['10th'].map('{:.2f}%'.format)
reading_grades_df_clean['11th']=reading_grades_df['11th'].map('{:.2f}%'.format)
reading_grades_df_clean['12th']=reading_grades_df['12th'].map('{:.2f}%'.format)
reading_grades_df_clean


9th    10th    11th    12th
Bailey High School     81.30%  80.91%  80.95%  80.91%
Cabrera High School    83.68%  84.25%  83.79%  84.29%
Figueroa High School   81.20%  81.41%  80.64%  81.38%
Ford High School       80.63%  81.26%  80.40%  80.66%
Griffin High School    83.37%  83.71%  84.29%  84.01%
Hernandez High School  80.87%  80.66%  81.40%  80.86%
Holden High School     83.68%  83.32%  83.82%  84.70%
Huang High School      81.29%  81.51%  81.42%  80.31%
Johnson High School    81.26%  80.77%  80.62%  81.23%
Pena High School       83.81%  83.61%  84.34%  84.59%
Rodriguez High School  80.99%  80.63%  80.86%  80.38%
Shelton High School    84.12%  83.44%  84.37%  82.78%
Thomas High School     83.73%  84.25%  83.59%  83.83%
Wilson High School     83.94%  84.02%  83.76%  84.32%
Wright High School     83.83%  83.81%  84.16%  84.07%

## Scores by School Spending

* Create a table that breaks down school performances based on average Spending Ranges (Per Student). Use 4 reasonable bins to group school spending. Include in the table each of the following:
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * Overall Passing Rate (Average of the above two)

In [32]:
#Setup bins and group names
bins = [0, 585, 615, 645, 675]
group_names = ["<$585", "$585-615", "$615-645", "$645-675"]

#Bin spending ranges
school_summ_df["Spending Ranges (Per Student)"] = pd.cut(school_summ_df["Per Student Budget"], bins, labels=group_names)

#Group by spending group
by_spending_group=school_summ_df.groupby('Spending Ranges (Per Student)')

#Get unique percentages
by_spending_math=by_spending_group['Average Math Score'].mean()
by_spending_reading=by_spending_group['Average Reading Score'].mean()
by_spending_math_pass=by_spending_group['% Passing Math'].mean()
by_spending_reading_pass=by_spending_group['% Passing Reading'].mean()
by_spending_overall_pass=by_spending_group['% Overall Passing Rate'].mean()

#Create Spending Range Data Frame
spending_summ_df = pd.DataFrame({"Average Math Score": by_spending_math,
                                "Average Reading Score": by_spending_reading,
                                "% Passing Math": by_spending_math_pass,
                                "% Passing Reading": by_spending_reading_pass,
                                "% Overall Passing Rate": by_spending_overall_pass})

#Format Data Frame
spending_summ_df_clean=spending_summ_df
spending_summ_df_clean['Average Math Score']=spending_summ_df['Average Math Score'].map('{:.2f}%'.format)
spending_summ_df_clean['Average Reading Score']=spending_summ_df['Average Reading Score'].map('{:.2f}%'.format)
spending_summ_df_clean['% Passing Math']=spending_summ_df['% Passing Math'].map('{:.2%}'.format)
spending_summ_df_clean['% Passing Reading']=spending_summ_df['% Passing Reading'].map('{:.2%}'.format)
spending_summ_df_clean['% Overall Passing Rate']=spending_summ_df['% Overall Passing Rate'].map('{:.2%}'.format)

spending_summ_df_clean

Average Math Score Average Reading Score  \
Spending Ranges (Per Student)                                            
<$585                                     83.46%                83.93%   
$585-615                                  83.60%                83.89%   
$615-645                                  79.08%                81.89%   
$645-675                                  77.00%                81.03%   

                              % Passing Math % Passing Reading  \
Spending Ranges (Per Student)                                    
<$585                                 93.46%            96.61%   
$585-615                              94.23%            95.90%   
$615-645                              75.67%            86.11%   
$645-675                              66.16%            81.13%   

                              % Overall Passing Rate  
Spending Ranges (Per Student)                         
<$585                                         95.04%  
$585-615                                      95.07%  
$615-645                                      80.89%  
$645-675                                      73.65%

## Scores by School Size

* Perform the same operations as above, based on school size.

In [34]:
# Create bins and group anmes
bins = [0, 1000, 2000, 5000]
group_names = ["Small (<1000)", "Medium (1000-2000)", "Large (2000-5000)"]

#Bin school sizes
school_summ_df["School Size"] = pd.cut(school_summ_df["Total Students"], bins, labels=group_names)

#Group by school size
by_size_group=school_summ_df.groupby('School Size')

#Get unique percentages
by_size_math=by_size_group['Average Math Score'].mean()
by_size_reading=by_size_group['Average Reading Score'].mean()
by_size_math_pass=by_size_group['% Passing Math'].mean()
by_size_reading_pass=by_size_group['% Passing Reading'].mean()
by_size_overall_pass=by_size_group['% Overall Passing Rate'].mean()

#Create Spending Range Data Frame
size_summ_df = pd.DataFrame({"Average Math Score": by_size_math,
                                "Average Reading Score": by_size_reading,
                                "% Passing Math": by_size_math_pass,
                                "% Passing Reading": by_size_reading_pass,
                                "% Overall Passing Rate": by_size_overall_pass})

#Format Data Frame
size_summ_df_clean=size_summ_df
size_summ_df_clean['Average Math Score']=size_summ_df['Average Math Score'].map('{:.2f}%'.format)
size_summ_df_clean['Average Reading Score']=size_summ_df['Average Reading Score'].map('{:.2f}%'.format)
size_summ_df_clean['% Passing Math']=size_summ_df['% Passing Math'].map('{:.2%}'.format)
size_summ_df_clean['% Passing Reading']=size_summ_df['% Passing Reading'].map('{:.2%}'.format)
size_summ_df_clean['% Overall Passing Rate']=size_summ_df['% Overall Passing Rate'].map('{:.2%}'.format)

size_summ_df_clean


Average Math Score Average Reading Score % Passing Math  \
School Size                                                                  
Small (<1000)                  83.82%                83.93%         93.55%   
Medium (1000-2000)             83.37%                83.86%         93.60%   
Large (2000-5000)              77.75%                81.34%         69.96%   

                   % Passing Reading % Overall Passing Rate  
School Size                                                  
Small (<1000)                 96.10%                 94.82%  
Medium (1000-2000)            96.79%                 95.20%  
Large (2000-5000)             82.77%                 76.36%

Average Math Score  Average Reading Score  % Passing Math  \
School Size                                                                     
Small (<1000)                83.821598              83.929843       93.550225   
Medium (1000-2000)           83.374684              83.864438       93.599695   
Large (2000-5000)            77.746417              81.344493       69.963361   

                    % Passing Reading  % Overall Passing Rate  
School Size                                                    
Small (<1000)               96.099437               94.824831  
Medium (1000-2000)          96.790680               95.195187  
Large (2000-5000)           82.766634               76.364998

## Scores by School Type

* Perform the same operations as above, based on school type.

In [35]:
#Group by school type
by_type_group=school_summ_df.groupby('School Type')

#Get unique percentages
by_type_math=by_type_group['Average Math Score'].mean()
by_type_reading=by_type_group['Average Reading Score'].mean()
by_type_math_pass=by_type_group['% Passing Math'].mean()
by_type_reading_pass=by_type_group['% Passing Reading'].mean()
by_type_overall_pass=by_type_group['% Overall Passing Rate'].mean()

#Create Spending Range Data Frame
type_summ_df = pd.DataFrame({"Average Math Score": by_type_math,
                                "Average Reading Score": by_type_reading,
                                "% Passing Math": by_type_math_pass,
                                "% Passing Reading": by_type_reading_pass,
                                "% Overall Passing Rate": by_type_overall_pass})

#Format Data Frame
type_summ_df_clean=type_summ_df
type_summ_df_clean['Average Math Score']=type_summ_df['Average Math Score'].map('{:.2f}%'.format)
type_summ_df_clean['Average Reading Score']=type_summ_df['Average Reading Score'].map('{:.2f}%'.format)
type_summ_df_clean['% Passing Math']=type_summ_df['% Passing Math'].map('{:.2%}'.format)
type_summ_df_clean['% Passing Reading']=type_summ_df['% Passing Reading'].map('{:.2%}'.format)
type_summ_df_clean['% Overall Passing Rate']=type_summ_df['% Overall Passing Rate'].map('{:.2%}'.format)

type_summ_df_clean


Average Math Score Average Reading Score % Passing Math  \
School Type                                                           
Charter                 83.47%                83.90%         93.62%   
District                76.96%                80.97%         66.55%   

            % Passing Reading % Overall Passing Rate  
School Type                                           
Charter                96.59%                 95.10%  
District               80.80%                 73.67%

#Conclusions
A number of conclusions can be drawn from the given data. We first observe the overall rates among the population as a whole, setting a baseline to compare the subsequent analyses. Right away, we see that the top 5 schools based on passing rate are charter schools, while the bottom 5 are public. In fact, charter schools have an overall passing rate of 95% whereas district schools have a rate of 74%. This is a huge gap, and more thorough analysis of the differences between the two types is needed to draw deeper conclusions. Another observation is that as a whole, schools with higher budgets, did not yield better test results. In fact, schools with the highest budget per student performed the worst compared to their cheaper couterparts, and this inverse relationship continues across each budget level. We also observe that smaller and medium sized schools dramatically out-performed large sized schools on passing rates. A more thorough analyis delineating the differences between type, size, and budget is needed to uncover the underlying cause of these disparities.
